lib/urlwatch/main.py

```diff
@@ -68,6 +68,20 @@ def __init__(self, urlwatch_config, config_storage, cache_storage, urls_storage)
         if hasattr(self.urlwatch_config, 'migrate_urls'):
             self.urlwatch_config.migrate_cache(self)
 
+    def should_run(self, idx, job):
+        if not job.is_enabled():
+            return False
+
+        if self.urlwatch_config.tags:
+            # If we're using tags, check the job matches the tags
+            return job.matches_tags(self.urlwatch_config.tag_set)
+        else:
+            # If we're using indexes, no indexes means all jobs
+            if self.urlwatch_config.idx_set:
+                return idx in self.urlwatch_config.idx_set
+            else:
+                return True
```

Comment:
When the last statement in an `if` block is a `return`, I like to save some indentation using:

```suggestion
        if self.urlwatch_config.tags:
            # If we're using tags, check the job matches the tags
            return job.matches_tags(self.urlwatch_config.tag_set)

        # If we're using indexes, no indexes means all jobs
        if self.urlwatch_config.idx_set:
            return idx in self.urlwatch_config.idx_set
            
        return True
```